src/state/session/index.tsx

```diff
@@ -363,16 +369,24 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
     async account => {
       logger.debug(`session: initSession`, {}, logger.DebugContext.session)
 
-      const agent = new BskyAgent({
-        service: account.service,
-        persistSession: createPersistSessionHandler(
+      const agent = new BskyAgent({service: account.service})
+
+      // restore the correct PDS URL if available
+      if (account.pdsUrl) {
+        agent.pdsUrl = agent.api.xrpc.uri = new URL(account.pdsUrl)
```

Comment:
We should probs make an API on the agent for this

Reply:
I was thinking of moving the pdsUrl to the session object and patching the `.xrpc.api.uri` into a getter for that, wdyt? I think I still have that draft I could pull up

Reply:
Opened PR https://github.com/bluesky-social/atproto/pull/2431

Reply:
Appreciate it. Going to be some big changes with the OAuth update; will try to roll it into that